src/core/dash.h

```diff
@@ -990,16 +990,15 @@ auto DashTable<_Key, _Value, Policy>::Traverse(Cursor curs, Cb&& cb) -> Cursor {
 
 template <typename _Key, typename _Value, typename Policy>
 auto DashTable<_Key, _Value, Policy>::AdvanceCursorBucketOrder(Cursor cursor) -> Cursor {
-  // We fix bid and go over all segments. Once we reach the end we increase bid and repeat.
+  // We fix bid and go over all segments. Once we reach the end we decrease bid and repeat.
   uint32_t sid = cursor.segment_id(global_depth_);
   uint8_t bid = cursor.bucket_id();
   sid = NextSeg(sid);
   if (sid >= segment_.size()) {
+    if (bid == 0)
+      return Cursor{global_depth_, 0, LargestBucketId()};  // "End of traversal" cursor.
```

Comment:
I find this implementation a bit hacky but if we put it aside for a second,

Are we still assuming that TraverseBuckets should provide full coverage for the items that exist at the moment of traversal start? Insertion can move items from the neighbour bucket to its neighbour (i.e. logical bid + 2). 
So it is possible that the traversal finished a bucket K and then a new item from K-1 will be added to it before it continues to K-1. See ` Segment<Key, Value, Policy>::InsertUniq`. Are you still relying on CVCUponInsert to defend yourselves against such moves?

Reply:
it can also move items from bid to `bid-1` and here as well we probably rely on CVCUponInsert to protect against these changes.

Reply:
> in fact, what happens if we finished with stash buckets in the segment but then an insertion adds a new item to them?

If we insert a new item to a bucket that was already serialized this is ok because we will have the journal change which will add this item in the replica.
The problem is when we move items from one bucket that was not serialized to another bucket that was serialized (as you noted in inset unique there is flow of MoveToOther which moves an item that was not serialized to a bucket which was serialized)

Reply:
so I suggest doing the same even if we do the full sync without snapshot isolation (and to keep the same ordering)